tensorflow/c/experimental/gradients/nn_grad.cc

```diff
@@ -110,6 +111,48 @@ class SparseSoftmaxCrossEntropyWithLogitsGradientFunction
   vector<AbstractTensorHandle*> forward_outputs;
 };
 
+// TODO(vnvo2409): Add python test
+class BiasAddGradientFunction : public GradientFunction {
+ public:
+  explicit BiasAddGradientFunction(AttrBuilder f_attrs)
+      : forward_attrs(f_attrs) {}
+
+  Status Compute(Context* ctx, const IncomingGradients& grad_inputs,
+                 vector<AbstractTensorHandle*>* grad_outputs) override {
+    /* Given upstream grad U and a BiasAdd: A + bias, the gradients are:
+     *
+     *    dA = U
+     *    dbias = reduceSum(U, dims = channel_dim)
+     */
+
+    AbstractTensorHandle* upstream_grad = grad_inputs[0];
+    DCHECK(upstream_grad);
+    grad_outputs->resize(2);
+
+    // Recover data format from forward pass for gradient.
+    std::string data_format;
+    forward_attrs.Get("data_format", &data_format);
```

Comment:
This fails internally. Change this to `TF_RETURN_IF_ERROR(forward_attrs.Get("data_format", &data_format));`

Reply:
Done